### PR TITLE
Survey added

### PR DESCRIPTION
**Pull request by ViraMaximets**

### CODE REVIEW

File: src/main/java/com/softserve/app/controller/CheckBoxController.java [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+@RestController
+@AllArgsConstructor
+@CrossOrigin(origins = "*")
+public class CheckBoxController {
+    private final CheckBoxService checkBoxService;
+    private final UserService userService;
+    private final SurveyService surveyService;
+
+    @GetMapping("/user/{user_id}/surveys/{survey_id}")
+    public List<CheckBox>  n(
```
Comment (Qvel) [inline src/main/java/com/softserve/app/controller/CheckBoxController.java:33]: dto?

Comment (Qvel) [reply src/main/java/com/softserve/app/controller/CheckBoxController.java:33]: method name?

File: src/main/java/com/softserve/app/controller/SurveyController.java [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+        return ResponseEntity.ok(SportHubConstant.SURVEY_DELETED.getMessage());
+    }
+
+    @GetMapping("/user/{user_id}/surveys/open")
+    public List<Survey> findAllOpen(
+            @PathVariable Long user_id) {
+        List<Survey> openSurveys = new ArrayList<>();
+        User usr = userService.findById(user_id);
+        List<Survey> surveys =  surveyService.findAllByUser(usr);
+        for (Survey s : surveys){
```
Comment (Qvel) [inline src/main/java/com/softserve/app/controller/SurveyController.java:66]: move logic to the service

File: src/main/java/com/softserve/app/controller/SurveyController.java [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+    }
+
+    @GetMapping("/user/{user_id}/surveys/closed")
+    public List<Survey> findAllClosed(
+            @PathVariable Long user_id) {
+        List<Survey> closedSurveys = new ArrayList<>();
+        User usr = userService.findById(user_id);
+        List<Survey> surveys =  surveyService.findAllByUser(usr);
+        for (Survey s : surveys){
+            if (!s.getIsOpen()) {
```
Comment (Qvel) [inline src/main/java/com/softserve/app/controller/SurveyController.java:81]: to the service

**Review by Qvel — CHANGES_REQUESTED** [review]

**Review by Qvel — CHANGES_REQUESTED** [review]

File: src/main/java/com/softserve/app/models/View.java [hunk 4]
```diff
@@ -0,0 +1,6 @@
+package com.softserve.app.models;
+
+
+public final class View {
```
[the remark this answers is not in the document]
Comment (ViraMaximets) [reply src/main/java/com/softserve/app/models/View.java:4]: i don`t ..

File: pom.xml [hunk 5]
```diff
@@ -102,6 +102,12 @@
             <scope>test</scope>
         </dependency>
 
+        <dependency>
+            <groupId>javax.annotation</groupId>
```
[the remark this answers is not in the document]
Comment (ViraMaximets) [reply pom.xml:106]: used for postController

File: src/main/java/com/softserve/app/constant/SportHubConstant.java [hunk 6]
```diff
@@ -23,22 +23,22 @@
             + "<br>"
             + "<p>Ignore this email if you do remember your password, "
             + "or you have not made the request.</p>"),
+    EMAIL_SENDING_EXCEPTION("Exception while performing email sending"),
 
 
     AUTHORIZE_EXCEPTION("Sorry, you're not authorized to access this resource"),
+    JWT_EXCEPTION("Unable to get JWT Token or JWT Token has expired"),
```
[the remark this answers is not in the document]
Comment (ViraMaximets) [reply src/main/java/com/softserve/app/constant/SportHubConstant.java:30]: this from old version of main..

File: src/main/java/com/softserve/app/dto/SurveyDTO.java [hunk 7]
```diff
@@ -0,0 +1,18 @@
...
+import lombok.AllArgsConstructor;
+import lombok.Builder;
+import lombok.Data;
+import lombok.NoArgsConstructor;
+
+@Data
+@NoArgsConstructor
+@AllArgsConstructor
+@Builder(toBuilder = true)
+@JsonIgnoreProperties(ignoreUnknown = true)
```
[the remark this answers is not in the document]
Comment (ViraMaximets) [reply src/main/java/com/softserve/app/dto/SurveyDTO.java:13]: important for angular

**Review by Qvel — APPROVED** [review]

**Pull request merged**